### DISCUSSION TITLE
Chai .should('be.enabled') doesn't seem to work in all cases.

### DISCUSSION THREAD

**Discussion by aagranovExtend** [cypress-io/cypress#26704] [Questions and Help]
I have a `<button aria-disabled="false".../>` that assertion `.should('be.enabled')` fails. If I change to use chai-JQuery `.should('have.attr', 'aria-disabled', 'false'`) that passes. It seems that the presence of the `aria-disabled` doesn't allow the normal Chai assertion to work.

Anyone else encounter this?

**Comment by portyu9** [comment]
Hi @aagranovExtend 

`aria-disabled` is not what Cypress's `be.enabled` / `be.disabled` assertion uses to determine the native enabled state.

These represent different things:

```html
<button aria-disabled="false">
```

and:

```html
<button disabled>
```

`aria-disabled` communicates an accessibility state. It does not set the HTML `disabled` property.

Therefore, a normal `<button>` with only:

```html
aria-disabled="false"
```

should still be natively enabled. If `.should('be.enabled')` is failing for that element, inspect its actual disabled state rather than assuming `aria-disabled` caused it:

```js
cy.get('button')
  .should('have.attr', 'aria-disabled', 'false')
  .and('have.prop', 'disabled', false)
  .and('be.enabled')
```

Also check whether it is inside a disabled `<fieldset>`, since that can make a control effectively disabled even though its own `disabled` property is false.

Conversely, if your application uses:

```html
<button aria-disabled="true">
```

without the native `disabled` attribute/property, Cypress may still consider the button natively enabled. In that case, test the ARIA contract explicitly:

```js
cy.get('button')
  .should('have.attr', 'aria-disabled', 'true')
```

and separately test whatever application logic prevents the action.

So the important point is: **ARIA state and native HTML disabled state should be asserted separately.**

There were no screenshots or follow-up comments in the discussion that change this conclusion.

Please mark this answer as accepted if it helped, thank you.
